tracer/src/Datadog.Trace/AppSec/Waf/Context.cs

```diff
@@ -36,6 +36,11 @@ public Context(IntPtr contextHandle, WafNative wafNative, Encoder encoder)
 
         public IResult Run(ulong timeoutMicroSeconds)
         {
+            if (disposed)
+            {
+                throw new Exception("Can't run WAF when disposed");
```

Comment:
If this _is_ disposed (somehow, due to a bug) then this will likely crash customers right? Should we log a warning and just return to be safe?

Reply:
Exceptions should never leak from the appsec code into the main tracer code or the users app. I believe this works well, as while working on this PR, I created a bug that tested this scenario 😄 

We could log and return, but throwing an exception provides a coinvent way of stopping further execution and logging in the handler that will catch the exception.

Reply:
Ah cool, nice 🙂